### PR TITLE
set quarantining and allow empty test results flags defaults to true

### PR DESCRIPTION
**Pull request by dfrankland**
https://trunk-io.slack.com/archives/C044VBASZ0D/p1732053725990549?thread_ts=1732053607.614599&cid=C044VBASZ0D

### CODE REVIEW

File: cli/src/upload.rs [hunk 1]
```diff
@@ -72,10 +72,19 @@ pub struct UploadArgs {
...
+        long,
+        help = "Run commands with the quarantining step.",
+        default_value = "true"
+    )]
     pub use_quarantining: bool,
-    #[arg(long, help = "Do not fail if no junit files are found.")]
-    pub allow_missing_junit_files: bool,
+    #[arg(
+        long,
+        alias = "allow-missing-junit-files",
```
Comment (dfrankland) [inline cli/src/upload.rs:83]: Adds this deprecated flag as a hidden alias:

<details>
<summary>

Click to view `upload` command `--help` text

</summary>

```text
Upload data to Trunk Flaky Tests

Usage: trunk flakytests upload [OPTIONS] --org-url-slug <ORG_URL_SLUG> --token <TOKEN>

Options:
      --junit-paths <JUNIT_PATHS>
          Comma-separated list of glob paths to junit files.
      --xcresult-path <XCRESULT_PATH>
          Path of xcresult directory
      --org-url-slug <ORG_URL_SLUG>
          Organization url slug.
      --token <TOKEN>
          Organization token. Defaults to TRUNK_API_TOKEN env var. [env: TRUNK_API_TOKEN=]
      --repo-root <REPO_ROOT>
          Path to repository root. Defaults to current directory.
      --repo-url <REPO_URL>
          Value to override URL of repository.
      --repo-head-sha <REPO_HEAD_SHA>
          Value to override SHA of repository head.
      --repo-head-branch <REPO_HEAD_BRANCH>
          Value to override branch of repository head.
      --repo-head-commit-epoch <REPO_HEAD_COMMIT_EPOCH>
          Value to override commit epoch of repository head.
      --tags <TAGS>
          Comma separated list of custom tag=value pairs.
      --print-files
          Print files which will be uploaded to stdout.
      --dry-run
          Run metrics CLI without uploading to API.
      --team <TEAM>
          Value to tag team owner of upload.
      --codeowners-path <CODEOWNERS_PATH>
          Value to override CODEOWNERS file or directory path.
      --use-quarantining
          Run commands with the quarantining step.
      --allow-empty-test-results
          Do not fail if test results are not found.
  -h, --help
          Print help
```

</details>

File: cli-tests/src/upload.rs [hunk 2]
```diff
@@ -33,7 +33,6 @@ async fn upload_bundle() {
         .env("GITHUB_JOB", "test-job")
         .args(&[
             "upload",
-            "--use-quarantining",
```
Comment (dfrankland) [inline cli-tests/src/upload.rs:36]: No longer need to specify since this is set by default

File: cli-tests/src/upload.rs [hunk 3]
```diff
@@ -242,29 +239,52 @@ async fn upload_bundle_empty_junit_paths() {
...
+            .env("TRUNK_PUBLIC_API_ADDRESS", &state.host)
+            .env("CI", "1")
+            .args(&args)
+            .assert();
+
+        assert = if matches!(flag, Flag::Off) {
+            assert.failure()
+        } else {
+            assert.success()
+        };
```
Comment (dfrankland) [inline cli-tests/src/upload.rs:285]: Tests that the flag is aliased to its old name and is set by default as well as verifying the CLI fails when the flag is turned off

**Review by TylerJang27 — APPROVED** [review]
LGTM

File: cli-tests/src/upload.rs (hunk 2 above)
Comment (federicoweber) [reply cli-tests/src/upload.rs:36]: We will still need a way to skip the quarantining step.

Comment (dfrankland) [reply cli-tests/src/upload.rs:36]: One may pass `--use-quarantining false` :)

**Pull request merged**